### PR TITLE
[LIB-751] Provide real test for CLI

### PR DESCRIPTION
**Pull request by opalczynski**

### CODE REVIEW

**Comment by opalczynski** [comment]
@23doors @Zhebr If would be nice if someone take a fresh look at this, I've a little mind distortion now :)

File: syncano_cli/main.py [hunk 1]
```diff
@@ -9,6 +9,10 @@
 from syncano_cli.parse_to_syncano.commands import top_migrate
 from syncano_cli.sync.commands import top_sync
 
+# sys.path.append('/home/sopalczy/projects/syncano/syncano-cli')
```
Comment (23doors) [inline syncano_cli/main.py]: You left something behind :P

Comment (opalczynski) [inline syncano_cli/main.py]: agrhrhrh :) sry, correcting.

File: tests/base.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        )
+
+    @classmethod
+    def tearDownClass(cls):
+        cls.connection = None
+
+    @classmethod
+    def generate_hash(cls):
+        hash_feed = '{}{}'.format(uuid4(), datetime.now())
+        return md5(hash_feed.encode('ascii')).hexdigest()
```
Comment (23doors) [inline tests/base.py:37]: hmac is a less likely to produce collisions but in this case it doesn't matter much

File: tests/base.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+
+class InstanceMixin(object):
+
+    @classmethod
+    def setUpClass(cls):
+        super(InstanceMixin, cls).setUpClass()
+
+        cls.instance = cls.connection.Instance.please.create(
+            name='i%s' % cls.generate_hash()[:10],
```
Comment (23doors) [inline tests/base.py]: maybe some more meaningful prefix to more easily distinguish these? `test_`?

Comment (opalczynski) [inline tests/base.py]: I've stole it from python LIB :) But yep - this is good idea.

File: tests/test_migrate_commands.py [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+    def test_parse(self):
+        result = self.runner.invoke(cli, args=['migrate', 'parse'], obj={})
+        self.assertIn('Aborted!', result.output)
+
+    @mock.patch('syncano_cli.parse_to_syncano.parse.connection.ParseConnection.request')
+    @mock.patch('syncano_cli.parse_to_syncano.migrations.transfer.SyncanoTransfer.transfer_objects')
+    @mock.patch('syncano_cli.parse_to_syncano.migrations.transfer.SyncanoTransfer.transfer_files')
+    @mock.patch('syncano_cli.parse_to_syncano.migrations.transfer.SyncanoTransfer.process_relations')
+    def test_class_migrations(self, relations_mock, files_mock, objects_mock, request_mock):
+        relations_mock.return_value = None
```
Comment (23doors) [inline tests/test_migrate_commands.py]: You can define return_value as an argument on `@mock.patch`,
e.g.
`@mock.patch('syncano_cli.parse_to_syncano.migrations.transfer.SyncanoTransfer.process_relations', return_value=None)`
Or if you don't care about asserting .called etc you can do something like that:
`@mock.patch('syncano_cli.parse_to_syncano.migrations.transfer.SyncanoTransfer.process_relations', mock.MagicMock(return_value=None))` - this way it won't be passed as an argument to function

File: tests/test_sync_commands.py [hunk 5]
```diff
@@ -0,0 +1,246 @@
...
+
+class BaseSyncCLITest(unittest.TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        super(BaseSyncCLITest, cls).setUpClass()
+        cls.yml_file = 'syncano.yml'
+        cls.scripts_dir = 'scripts/'
+
+    def _assert_file_exists(self, path):
```
Comment (23doors) [inline tests/test_sync_commands.py]: Why are all these asserts private (or well, prefixed with '_' as concept of private doesn't exactly exist in python)?
They're completely valid and useful asserts. I think they should be public just like other assert methods on testcase class are.

**Comment by 23doors** [comment]
lgtm. other than a few comments, it looks solid! :+1:

**Pull request merged**